src/SmoFacade/BackupFileTools.cs

```diff
@@ -64,5 +64,17 @@ public static bool IsValidFilePath(string path)
         return false;
       }
     }
+
+    public static string CombinePaths(string path1, string path2)
+    {
+      if (string.IsNullOrEmpty(path1)) { return path2; }
+      if (string.IsNullOrEmpty(path2)) { return path1; }
+
+      // assumes path1 contains desired separators
+      var separator = path1.Contains(@"\") ? @"\" : "/";
+      return path1.EndsWith(separator) && path2.StartsWith(separator) ? 
+        $"{path1}{path2.Substring(1)}" :
+        path1.EndsWith(separator) || path2.StartsWith(separator) ? $"{path1}{path2}" : $"{path1}{separator}{path2}";
```

Comment:
I think all of this _will work_, why not make a simple fix:
```C#
var newFilePath = Path.Combine(path, fileName);
if(newFilePath .EndsWith("/\"))
  newFilePath.Remove(newFilePath.Length() - 1, 1);
```

Reply:
The substring `"/\"` would appear in the middle of the string so the logic would have to be something like 

```c#
var newFilePath = Path.Combine(path, fileName);
if(newFilePath.Contains("/\"))
{
  var start = /** Location of "/" **/;
  newFilePath.Remove(start, 1);
}
```

That's fine too. I think I would just prefer to replace the problematic method than code around it. I could simplify it further and just use `"/"` instead of programmatically determining the `separator`.

Reply:
Ah, right. why not do this then:
`newFilePath.Replace("/\", "/");`

Reply:
> I could simplify it further and just use "/" instead of programmatically determining the separator.

This means we won't be able to support windows paths.

Reply:
Oops, I meant the "\" character haha.

Reply:
I'd rather replace `Path.*` altogether than continue to use it. It doesn't work as intended on linux, and it seems a couple test cases fail for `ValidUrlTests` and `ValidPathsTests`.

I'm not even sure using `Path.*` is appropriate here. It's intended for cross-platform file path manipulation, but we're manipulating files on a remote machine, not the one this code is running on.

Reply:
ya that makes sense. 
DotNet has different implementations for the same method depending on which OS platform the code is running in.
Becomes very difficult when handling paths from both OSs.

The only thing is, I would suggest to quickly look online for ways people are handling this before we go with one way or the other. There might be an easier solution while still using `Path.*`